swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -399,89 +256,52 @@ extension Adapter: CallbackHandlerDelegate {
       self.logger.log("Adapter.onSetInterfaceConfig")
 
       switch self.state {
-      case .startingTunnel:
-        self.networkSettings = NetworkSettings(
-          tunnelAddressIPv4: tunnelAddressIPv4, tunnelAddressIPv6: tunnelAddressIPv6,
-          dnsAddresses: dnsAddresses)
-      case .tunnelReady:
-        if let networkSettings = self.networkSettings {
-          networkSettings.apply(
-            on: packetTunnelProvider,
-            logger: self.logger,
-            completionHandler: nil
-          )
+      case .startingTunnel(let session, let onStarted):
+        networkSettings.tunnelAddressIPv4 = tunnelAddressIPv4
+        networkSettings.tunnelAddressIPv6 = tunnelAddressIPv6
+        networkSettings.dnsAddresses = dnsAddresses
+        // Add DNS sentinels to routes as a start
```

Comment:
@conectado We can use this convention in all clients: add the DNS sentinels to routes when processing the `onSetInterfaceConfig` callback.

Or we can just defer that until `onUpdateRoutes`

Reply:
I added the same logic to Android. By ensuring the DNS server addresses are included in the tunnel routes in `onSetInterfaceConfig`, we avoid hiccups in resolving non-Firezone names until `onUpdateRoutes` is called.

Reply:
I'm not married to this approach, however. If you want to keep the `onAddRoute` and `onRemoveRoute` instead that's fine, but I may suggest to optimize it by making it multiple, e.g. `onAddRoutes` so that we only need to handle the callback once in the clients.

Reply:
I'll use `onUpdateRoutes` and discard `onAddRoute` and `onRemoveRoute` this moves us toward the direction of #3343 and from connlib perspective it's almost the same to implement.